### PR TITLE
[X86] Use X86AS::GS and X86AS::FS instead of 256 and 257. NFC

### PR DESCRIPTION
**Pull request by topperc**

### CODE REVIEW

File: clang/lib/CodeGen/CMakeLists.txt [hunk 1]
```diff
@@ -144,6 +144,7 @@ add_clang_library(clangCodeGen
   VarBypassDetector.cpp
 
   DEPENDS
+  vt_gen
```
Comment (phoebewang) [inline clang/lib/CodeGen/CMakeLists.txt:147]: Why it depends?

Comment (topperc) [reply clang/lib/CodeGen/CMakeLists.txt:147]: I don't know how that got in here.

**Review by phoebewang — APPROVED** [review]
LGTM.

File: clang/lib/CodeGen/CMakeLists.txt (hunk 1 above)
Comment (topperc) [reply clang/lib/CodeGen/CMakeLists.txt]: That change is c3fe727181818d3efdd2ce96598bfe6a2d3ffb83, but I'm not sure how it got squashed into my commit

**Pull request merged**